android/app/src/main/java/com/microsoft/codepush/react/CodePush.java

```diff
@@ -295,7 +295,16 @@ void initializeUpdateAfterRestart() {
 
         JSONObject pendingUpdate = mSettingsManager.getPendingUpdate();
         if (pendingUpdate != null) {
-            JSONObject packageMetadata = this.mUpdateManager.getCurrentPackage();
+            JSONObject packageMetadata = null;
+
+            try {
+                packageMetadata = this.mUpdateManager.getCurrentPackage();
+            } catch (CodePushMalformedDataException e) {
+                // We need to recover the app in case 'codepush.json' is corrupted
+                CodePushUtils.log(e.getMessage());
```

Comment:
`CodePushUtils.log(e)` might be better here as it will be logged as an exception.

Reply:
yep good point, fixed